lib/osutil/lowprio_linux.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2017 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package osutil
+
+import (
+	"os"
+	"syscall"
+)
+
+const ioprioClassShift = 13
+
+type ioprioClass int
+
+const (
+	ioprioClassRT ioprioClass = iota + 1
+	ioprioClassBE
+	ioprioClassIdle
+)
+
+const (
+	ioprioWhoProcess = iota + 1
+	ioprioWhoPGRP
+	ioprioWhoUser
+)
+
+func ioprioSet(class ioprioClass, value int) {
+	// error return ignored
+	syscall.Syscall(syscall.SYS_IOPRIO_SET,
+		uintptr(ioprioWhoPGRP), uintptr(os.Getpid()),
+		uintptr(class)<<ioprioClassShift|uintptr(value))
+}
+
+// SetLowPriority lowers the process CPU scheduling priority, and possibly
+// I/O priority depending on the platform and OS.
+func SetLowPriority() {
+	// Move ourselves to a new process group so that we can use the process
+	// group variants of Setpriority etc to affect all of our threads in one
+	// go. If this fails, bail, so that we don't affect things we shouldn't.
+	if err := syscall.Setpgid(0, 0); err != nil {
+		return
```

Comment:
sync.Once this?

Reply:
I don't see the point, it's just called once from main and we'd need to remember the error in another package variable to be able to return it again etc

Reply:
Yeah, I was just pointing out in case other Go users use it. Also, with errors being returned, sync.Once makes more sense (as I assume doing pgrp twice will fail).